src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineHelper.cs

```diff
@@ -0,0 +1,316 @@
+                method: Methods.TextDocumentDocumentSymbolName,
+                capabilitiesFilter: (JToken x) => true,
+                languageServerName: WellKnownLspServerKinds.AlwaysActiveVSLspServer.ToUserVisibleString(),
+                parameterFactory: _ => JToken.FromObject(parameterFactory),
+                cancellationToken: cancellationToken).ConfigureAwait(false);
+        }
+
+        public static void SetIsExpanded(ImmutableArray<DocumentSymbolItem> documentSymbolItems, ExpansionOption expansionOption)
+        {
+            foreach (var documentSymbolItem in documentSymbolItems)
+            {
+                documentSymbolItem.IsExpanded = expansionOption is ExpansionOption.Expand;
+                SetIsExpanded(documentSymbolItem.Children, expansionOption);
+            }
+        }
+
+        /// <summary>
+        /// Sorts and returns an immutable array of DocumentSymbolItem based on a SortOption.
+        /// </summary>
+        public static ImmutableArray<DocumentSymbolItem> Sort(
+            ImmutableArray<DocumentSymbolItem> documentSymbolItems,
+            SortOption sortOption,
+            CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            // Log which sort option was used
+            Logger.Log(sortOption switch
+            {
+                SortOption.Name => FunctionId.DocumentOutline_SortByName,
+                SortOption.Order => FunctionId.DocumentOutline_SortByOrder,
+                SortOption.Type => FunctionId.DocumentOutline_SortByType,
+                _ => throw new NotImplementedException(),
+            });
+
+            return SortDocumentSymbolItems(documentSymbolItems, sortOption, cancellationToken);
+
+            static ImmutableArray<DocumentSymbolItem> SortDocumentSymbolItems(
+                ImmutableArray<DocumentSymbolItem> documentSymbolItems,
+                SortOption sortOption,
+                CancellationToken cancellationToken)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                // Sort the top-level DocumentSymbolItems
+                var sortedDocumentSymbolItems = sortOption switch
+                {
+                    SortOption.Name => documentSymbolItems.Sort((x, y) => x.Name.CompareTo(y.Name)),
+                    SortOption.Order => documentSymbolItems.Sort((x, y) =>
+                    {
+                        if (x.StartPosition.Line == y.StartPosition.Line)
+                            return x.StartPosition.Character - y.StartPosition.Character;
+
+                        return x.StartPosition.Line - y.StartPosition.Line;
+                    }),
+                    SortOption.Type => documentSymbolItems.Sort((x, y) =>
+                    {
+                        if (x.SymbolKind == y.SymbolKind)
+                            return x.Name.CompareTo(y.Name);
+
+                        return x.SymbolKind - y.SymbolKind;
```

Comment:
interesting.  so we just sort by numeric kind.  is that what we want?  can we doc this?